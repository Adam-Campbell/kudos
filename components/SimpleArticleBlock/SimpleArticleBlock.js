import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as styleConstants from '../styleConstants';
import Link from 'next/link';
import { AnchorButton } from '../Button';

const BlockContainer = styled.div`
    box-shadow: 1px 1px 4px 2px ${styleConstants.colorShadow};
    padding: 16px;
    margin-bottom: 16px;
`;

const ArticleTitle = styled.p`
    font-family: ${styleConstants.fontPrimary};
    font-size: 24px;
    color: ${styleConstants.colorBodyText};
`;

const ArticleDescription = styled.p`
    font-family: ${styleConstants.fontSecondary};
    font-size: 16px;
    font-weight: 300;
    color: ${styleConstants.colorBodyText};
    line-height: 1.4;
`;


const SimpleArticleBlock = props => (
    <BlockContainer>
        <ArticleTitle>{props.articleTitle}</ArticleTitle>
        <ArticleDescription>{props.articleDescription}</ArticleDescription>
        <Link passHref as={`/post/${props.article_id}`} href={`/post?post=${props.article_id}`}>
            <AnchorButton>View Article</AnchorButton>
        </Link>
    </BlockContainer>
);

SimpleArticleBlock.propTypes = {
    articleTitle: PropTypes.string.isRequired,
    articleDescription: PropTypes.string.isRequired,
    article_id: PropTypes.string.isRequired
};

export default SimpleArticleBlock;