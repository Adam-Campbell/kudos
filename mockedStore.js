export const store = {
    articles: {
        article_A: {
            _id: 'article_A',
            title: 'The title for Article A',
            description: 'The description for Article A.',
            image: 'https://example.com/article_a.jpg',
            text: 'The main body of Article A.',
            kudos: 7,
            author: 'user_A',
            category: 'category_A',
            commentIds: ['comment_A', 'comment_C']
        },
        article_B: {
            _id: 'article_B',
            title: 'The title for Article B',
            description: 'The description for Article B.',
            image: 'https://example.com/article_b.jpg',
            text: 'The main body of Article B.',
            kudos: 2,
            author: 'user_B',
            category: 'category_B',
            commentIds: ['comment_B', 'comment_D']
        },
        article_C: {
            _id: 'article_C',
            title: 'The title for Article C',
            description: 'The description for Article C.',
            image: 'https://example.com/article_c.jpg',
            text: 'The main body of Article C.',
            kudos: 0,
            author: 'user_A',
            category: 'category_B',
            commentIds: []
        }
    },
    users: {
        user_A: {
            _id: 'user_A',
            followers: 1,
            following: 0,
            username: 'Joe Bloggs',
            avatar: 'https://example.com/user_a.jpg',
            bio: 'The bio for User A.',
            postIds: ['article_A', 'article_C'],
            commentIds: ['comment_A', 'comment_B'],
            kudosIds: ['article_A'],
            highlightIds: ['article_A']
        },
        user_B: {
            _id: 'user_B',
            followers: 0,
            following: 1,
            username: 'Jane Smith',
            avatar: 'https://example.com/user_b.jpg',
            bio: 'The bio for User B.',
            postIds: ['article_B'],
            commentIds: ['comment_C'],
            kudosIds: ['article_B']
        }
    },
    comments: {
        comment_A: {
            _id: 'comment_A',
            text: 'The text for Comment A.',
            author: 'user_A',
            discussion: 'article_A',
            createdAt: 'Wed, 06 Jun 2018 08:07:53 GMT',
            parents: ['comment_A']
        },
        comment_B: {
            _id: 'comment_B',
            text: 'The text for Comment B.',
            author: 'user_A',
            discussion: 'article_B',
            createdAt: 'Wed, 06 Jun 2018 10:32:06 GMT',
            parents: ['comment_B']
        },
        comment_C: {
            _id: 'comment_C',
            text: 'The text for Comment C.',
            author: 'user_B',
            discussion: 'article_A',
            createdAt: 'Wed, 06 Jun 2018 12:21:27 GMT',
            parents: ['comment_A', 'comment_B']
        },
        comment_D: {
            _id: 'comment_D',
            text: 'Comment deleted by user.',
            discussion: 'article_B',
            createdAt: 'Wed, 06 Jun 2018 13:46:12 GMT',
            author: null,
            discussion: 'article_B',
            parents: ['comment_D']
        }
    },
    highlights: {
        highlight_A: {
            _id: 'highlight_A',
            post: 'article_A',
            user: 'user_A',
            excerpt: 'An excerpt from Article A.'
        }
    },
    currentUser: {
        isLoggedIn: true,
        token: 'abc123',
        _id: 'user_A',
        follows: ['user_B'],
        email: 'example@example.com'
    },
    categories: {
        all: ['article_A', 'article_B', 'article_C'],
        category_A: ['article_A'],
        category_B: ['article_B', 'article_C']
    },
    errors: {
        signUpDuplicateError: false,
        emailNotFound: false
    },
    successes: {
        passwordResetEmailSent: false
    }
}