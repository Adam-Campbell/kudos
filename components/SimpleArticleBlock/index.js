import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SimpleArticleBlock from './SimpleArticleBlock';

export const SimpleArticleBlockContainer = props => {
    const article = props.articles[props.article_id];
    return <SimpleArticleBlock 
        articleTitle={article.title}
        articleDescription={article.description}
        article_id={article._id}
    />
};

SimpleArticleBlockContainer.propTypes = {
    article_id: PropTypes.string.isRequired
}

const mapStateToProps = state => ({
    articles: state.posts.models
});

export default connect(
    mapStateToProps
)(SimpleArticleBlockContainer);