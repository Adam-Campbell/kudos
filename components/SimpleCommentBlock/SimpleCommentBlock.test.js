import { shallow } from 'enzyme';
import { store } from '../../mockedStore';
import { SimpleCommentBlockContainer } from './index';
import { SimpleArticleBlockContainer } from '../SimpleArticleBlock/index';
const comment_id = 'comment_A'

describe('<SimpleCommentBlockContainer/>', () => {
    test('renders correctly', () => {
        const container = shallow(
            <SimpleCommentBlockContainer 
                comment_id={comment_id}
                comments={store.comments}
            />
        );
        const presenter = container.first().shallow();
        expect(container).toMatchSnapshot();
        expect(presenter).toMatchSnapshot();
    });
});