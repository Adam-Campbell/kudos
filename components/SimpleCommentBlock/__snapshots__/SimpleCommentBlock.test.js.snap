// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleCommentBlockContainer/> renders correctly 1`] = `
<SimpleCommentBlock
  comment_id="comment_A"
  discussion_id="article_A"
/>
`;

exports[`<SimpleCommentBlockContainer/> renders correctly 2`] = `
<BlockContainer>
  <Connect(CommentDisplay)
    comment_id="comment_A"
  />
  <Link
    as="/post/article_A"
    href="/post?post=article_A"
    passHref={true}
  >
    <DiscussionAnchor>
      View Discussion
    </DiscussionAnchor>
  </Link>
</BlockContainer>
`;
